src/app/Dashboard/DraggableRef.tsx

```diff
@@ -35,125 +35,327 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE.
  */
-import {
-  CardConfig,
-  dashboardCardConfigResizeCardIntent,
-} from '@app/Shared/Redux/Configurations/DashboardConfigSlicer';
-import { RootState } from '@app/Shared/Redux/ReduxStore';
-import { gridSpans } from '@patternfly/react-core';
-import _ from 'lodash';
+import { dashboardCardConfigReorderCardIntent } from '@app/Shared/Redux/Configurations/DashboardConfigSlicer';
+import { css } from '@patternfly/react-styles';
 import React from 'react';
-import { useDispatch, useSelector } from 'react-redux';
-import { DashboardCardSizes } from './Dashboard';
-import { DashboardCardContext } from './ResizableCard';
+import { useDispatch } from 'react-redux';
+
+type itemPosition = 'left' | 'right' | 'else';
+//
+// |-----------------| mouse |-----------------|
+// |   rect1.right   ^       ^   rect2.left    |
+// |                 |       |                 |
+// returns [inBetweenTwoRectangles, {insertedOnLeft, insertedOnRight, else}]
+function inBetween(ev: MouseEvent, rect1: DOMRect, rect2: DOMRect): [boolean, itemPosition] {
```

Comment:
Use `const a = () => {}` to avoid hoisting? Not a problem but just easier to read/debug.